src/picomsg/src/oper_db_writer.cpp

```diff
@@ -25,6 +25,97 @@ static bool terminator_state_to_oper_db(const pm_terminator_conn_state terminato
 	return false;
 }
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wswitch-enum"
```

Comment:
I prefer to add such pragmas only on the specific functions that need it (and only around the relevant switch-case). Otherwise it's very easy to add a function in between these pragmas without knowing of the warning being ignored (think of having 100s of lines between the "push" and "pop"), and so this just crippled a warning for this function that might actually need it.